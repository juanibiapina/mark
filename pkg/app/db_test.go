package app

import (
	"os"
	"path"
	"testing"

	"mark/pkg/model"

	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/stretchr/testify/require"
)

func TestFilesystemDatabase(t *testing.T) {
	t.Parallel()

	t.Run("SaveConversation", func(t *testing.T) {
		// given
		dir := t.TempDir()
		c := model.MakeConversation()
		db := MakeFilesystemDatabase(dir)

		// when
		err := db.SaveConversation(c)
		require.Nil(t, err)

		// then
		filename := c.ID + ".json"
		file := path.Join(dir, filename)
		content, err := os.ReadFile(file)
		require.Nil(t, err)

		snaps.MatchSnapshot(t, string(content))
	})
}
