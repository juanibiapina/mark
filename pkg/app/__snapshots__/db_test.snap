
[TestFilesystemDatabase/SaveConversation - 1]
{"id":"2025-02-17T22:09:24.592607+01:00","prompt":null,"messages":null}
---
